server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -211,6 +212,31 @@ public static <E> List<E> appendToCopy(Collection<E> collection, E element) {
         return Collections.unmodifiableList(Arrays.asList(array));
     }
 
+    /**
+     * Creates a copy of the given set and adds extra element.
+     *
+     * @param set     set to copy
+     * @param element element to add
+     */
+    public static <E> Set<E> addToCopy(Set<E> set, E element) {
```

Comment:
Wouldn't it be possible to implement this just with the array in the following, and just loop through the `elements`?

Reply:
I introduced this similarly to `Set.of(...)` to avoid array creation when only a single element needs to be added.